modules/python/python-persist.c

```diff
@@ -127,3 +161,272 @@ python_format_persist_name(const LogPipe *p, const gchar *module, PythonPersistM
 
   return persist_name;
 }
+
+static PyObject *
+_persist_type_new(PyTypeObject *type, PyObject *args, PyObject *kwds)
+{
+  PyPersist *self =(PyPersist *)type->tp_alloc(type, 0);
+
+  GlobalConfig *cfg = main_loop_get_current_config(main_loop_get_instance());
+  self->persist_state = cfg->state;
+
```

Comment:
Is there a reason this is in new() and not in init()?

Reply:
I moved it ot init. As tp_new is not needed anymore, I changed it to `PyType_GenericNew`.